### PR TITLE
HostFS: Add remove file functionality

### PR DESCRIPTION
**Pull request by fjtrujy**
### Description of Changes
This PRs just add the remove file functionallity over the IOPBIOS, sorry guys I forgot to create it.

Thanks!

### CODE REVIEW

File: pcsx2/IopBios.cpp [hunk 1]
```diff
@@ -924,6 +939,7 @@ namespace R3000A
 			EXPORT_H(  6, read)
 			EXPORT_H(  7, write)
 			EXPORT_H(  8, lseek)
+            EXPORT_H( 10, remove)
```
Comment (refractionpcsx2) [inline pcsx2/IopBios.cpp:942]: can you fix the tabs vs spaces please :)

File: pcsx2/IopBios.cpp [hunk 2]
```diff
@@ -611,6 +611,21 @@ namespace R3000A
...
 
+		int remove_HLE()
+		{
+			const std::string full_path = clean_path(Ra0);
+
+			if (is_host(full_path))
+			{
+				const std::string path = full_path.substr(full_path.find(':') + 1);
+				ghc::filesystem::path file_path{host_path(path)};
+				bool suceeds = ghc::filesystem::remove(file_path);
```
Comment (lightningterror) [inline pcsx2/IopBios.cpp:622]: Can also be marked as const?

Comment (fjtrujy) [reply pcsx2/IopBios.cpp:622]: Should we do it then in the rest of places in this file?

Comment (refractionpcsx2) [reply pcsx2/IopBios.cpp:622]: sure, why not :)

Comment (fjtrujy) [reply pcsx2/IopBios.cpp]: Done!

File: pcsx2/IopBios.cpp [hunk 3]
```diff
@@ -657,8 +672,8 @@ namespace R3000A
 			if (is_host(full_path))
 			{
 				const std::string path = full_path.substr(full_path.find(':') + 1);
-				ghc::filesystem::path folder_path{host_path(path)};
-				bool suceeds = ghc::filesystem::remove(folder_path);
+				const ghc::filesystem::path folder_path{host_path(path)};
+				const bool suceeds = ghc::filesystem::remove(folder_path);
```
Comment (refractionpcsx2) [inline pcsx2/IopBios.cpp:676]: typo, succeeds would be correct, but succeeded might be a better way to word it?

File: pcsx2/IopBios.cpp [hunk 4]
```diff
@@ -611,15 +611,30 @@ namespace R3000A
...
 		{
 			const std::string full_path = clean_path(Ra0);
 
 			if (is_host(full_path))
 			{
 				const std::string path = full_path.substr(full_path.find(':') + 1);
-				ghc::filesystem::path folder_path{host_path(path)};
-				bool suceeds = ghc::filesystem::create_directory(folder_path);
+				const ghc::filesystem::path folder_path{host_path(path)};
+				const bool suceeds = ghc::filesystem::create_directory(folder_path);
```
Comment (refractionpcsx2) [inline pcsx2/IopBios.cpp:637]: typo, succeeds would be correct, but succeeded might be a better way to word it?

File: pcsx2/IopBios.cpp [hunk 5]
```diff
@@ -611,15 +611,30 @@ namespace R3000A
...
 
+		int remove_HLE()
+		{
+			const std::string full_path = clean_path(Ra0);
+
+			if (is_host(full_path))
+			{
+				const std::string path = full_path.substr(full_path.find(':') + 1);
+				const ghc::filesystem::path file_path{host_path(path)};
+				const bool suceeds = ghc::filesystem::remove(file_path);
```
Comment (refractionpcsx2) [inline pcsx2/IopBios.cpp]: typo, succeeds would be correct, but succeeded might be a better way to word it?

File: pcsx2/IopBios.cpp (hunk 3 above)
Comment (fjtrujy) [reply pcsx2/IopBios.cpp]: Applied!

File: pcsx2/IopBios.cpp (hunk 4 above)
Comment (fjtrujy) [reply pcsx2/IopBios.cpp]: Applied

**Review by refractionpcsx2 — APPROVED** [review]

**Review by tadanokojin — APPROVED** [review]
LGTM

**Comment by refractionpcsx2** [comment]
LGTMT, wait for builds to complete, then will merge

**Pull request merged**